### PR TITLE
 Implementing eval_as_leading_term for the Si function

### PR DESCRIPTION
**Pull request by Abhishekjsr283204**
References to other Issues or PRs

Brief description of what is fixed or changed

Adding a missing as_leading_term method for Si function 

Other comments

Release Notes
series
-> Implemented leading term methods for and Si function.

### CODE REVIEW

File: sympy/functions/special/error_functions.py [hunk 1]
```diff
@@ -1895,6 +1895,29 @@ def _eval_rewrite_as_sinc(self, z, **kwargs):
...
+            arg0 = arg.limit(x, 0, dir='-' if re(cdir).is_negative else '+')
+        if arg0.is_zero:
+            return arg
+        elif not arg0.is_infinite:
+            return self.func(arg0)
+        else:
+            return self
+    
+
+    def test_si():
```
Comment (anutosh491) [inline sympy/functions/special/error_functions.py:1912]: Wait I'm confused.  The leading term implementation looks correct to me but then the relevant tests should be added to the test file (`test_error_functions.py` I reckon) . Why have the tests been added here ?

File: sympy/functions/special/error_functions.py [hunk 2]
```diff
@@ -1895,6 +1895,29 @@ def _eval_rewrite_as_sinc(self, z, **kwargs):
...
+      assert Si(I*x) == I*Shi(x)
+      assert Shi(I*x) == I*Si(x)
+      assert Si(-I*x) == -I*Shi(x)
+      assert Shi(-I*x) == -I*Si(x)
+      assert Si(-x) == -Si(x)
+
```
Comment (anutosh491) [inline sympy/functions/special/error_functions.py:1918]: And these tests don't look relevant to your implementation at all. I think you would want to test out the `as_leading_term` method for `Si` function if I am not wrong

**Comment by anutosh491** [comment]
Do the following 
1) Add release notes for you changes.
2) Get rid of whitespaces so that the code quality test can pass.
3) Add your credentials to the mailmap file

File: sympy/functions/special/error_functions.py [hunk 3]
```diff
@@ -1895,6 +1895,29 @@ def _eval_rewrite_as_sinc(self, z, **kwargs):
         t = Symbol('t', Dummy=True)
         return Integral(sinc(t), (t, 0, z))
 
+    def _eval_as_leading_term(self, x, logx=None, cdir=0):
+        arg = self.args[0].as_leading_term(x)
```
Comment (jksuom) [inline sympy/functions/special/error_functions.py:1899]: This also depends on `cdir` and possibly on `logx` in some cases.

**Comment by anutosh491** [comment]
Could you please tests for this PR.
Also make sure all tests pass.
Also the PR description is very confusing. Please address that too.

**Comment by Abhishekjsr283204** [comment]
okay I will do

File: sympy/functions/special/error_functions.py (hunk 3 above)
Comment (jksuom) [reply sympy/functions/special/error_functions.py:1899]: `logx` and `cdir` should be added.

**Comment by anutosh491** [comment]
Please resolve the merge conflicts for the same !

File: sympy/series/tests/test_limits.py [hunk 4]
```diff
@@ -19,7 +19,7 @@
                                                       atan, cos, cot, csc, sec, sin, tan)
 from sympy.functions.special.bessel import (besseli, bessely, besselj, besselk)
 from sympy.functions.special.error_functions import (Ei, erf, erfc, erfi, fresnelc, fresnels)
-from sympy.functions.special.gamma_functions import (digamma, gamma, uppergamma)
+from sympy.functions.special.gamma_functions import (digamma, gamma, lowergamma, uppergamma)
 from sympy.functions.special.hyper import meijerg
```
Comment (anutosh491) [inline sympy/series/tests/test_limits.py:23]: Remove this

File: sympy/functions/special/gamma_functions.py [hunk 5]
```diff
@@ -555,6 +561,10 @@ def _eval_rewrite_as_tractable(self, s, x, **kwargs):
     def _eval_rewrite_as_expint(self, s, x, **kwargs):
         from sympy.functions.special.error_functions import expint
         return expint(1 - s, x)*x**s
+    
+    def _eval_as_leading_term(self, x, logx=None, cdir=0):
+        from sympy.functions.special.error_functions import expint
+        return self.rewrite(expint)._eval_as_leading_term(x, logx=logx, cdir=cdir)
```
Comment (anutosh491) [inline sympy/functions/special/gamma_functions.py:567]: Restore this file

File: sympy/functions/special/error_functions.py [hunk 6]
```diff
@@ -1414,6 +1414,18 @@ def _eval_rewrite_as_Si(self, nu, z, **kwargs):
...
+        from sympy.functions.special.gamma_functions import digamma
+        nu, z = self.args
+        if nu.is_real and nu >= 1:
+            term_one = (-z)**(nu - 1)/factorial(nu - 1) * (digamma(nu) - log(z))
+            term_two = 1/(nu - 1)
+            if nu == 1:
+                term_two = z/(2 - nu)
+            arg = Add(*[term_one, term_two]).as_leading_term(x, logx=logx)
+            return arg
+        return super(expint, self)._eval_as_leading_term(x, logx, cdir)
```
Comment (anutosh491) [inline sympy/functions/special/error_functions.py:1428]: Let's just use this PR to implement the _eval_as_leading_term function for the Si function. We shall look into other things later.

File: sympy/functions/special/tests/test_error_functions.py [hunk 7]
```diff
@@ -624,14 +624,15 @@ def test_si():
     assert Si(x).series(x, oo) == pi/2 - (- 6/x**3 + 1/x \
         + O(x**(-7), (x, oo)))*sin(x)/x - (24/x**4 - 2/x**2 + 1 \
         + O(x**(-7), (x, oo)))*cos(x)/x
-
+    
     t = Symbol('t', Dummy=True)
```
Comment (anutosh491) [inline sympy/functions/special/tests/test_error_functions.py:628]: Remove this whitespace here.

File: sympy/functions/special/tests/test_error_functions.py [hunk 8]
```diff
@@ -624,14 +624,15 @@ def test_si():
...
-    assert Si(x).rewrite(sinc).dummy_eq(Integral(sinc(t), (t, 0, x)))
-
```
Comment (anutosh491) [inline sympy/functions/special/tests/test_error_functions.py:630]: Okay wait, I'm not sure why have you removed this test. If this is already present in master, no need to disturb it

File: sympy/functions/special/tests/test_error_functions.py [hunk 9]
```diff
@@ -624,14 +624,15 @@ def test_si():
...
+    assert Si(x).rewrite(sinc) == Integral(sinc(t), (t, 0, x))
+    assert Si(x)._eval_as_leading_term(x) == x
+    assert Si(2*x)._eval_as_leading_term(x) == 2*x
+    assert Si(sin(x))._eval_as_leading_term(x) == x
```
Comment (anutosh491) [inline sympy/functions/special/tests/test_error_functions.py:632]: Add a case for Si(1/x) or Si(-1/x) and check if that works

**Comment by anutosh491** [comment]
Fix the code quality here and get the CI passing. Should be good to go after that
